### PR TITLE
spdm_crypt_lib: Update the API to handle cert_mode

### PR DESCRIPTION
**Pull request by sunceping**
Fix the issue: https://github.com/DMTF/libspdm/issues/2505

### CODE REVIEW

**Comment by Wenxing-hou** [comment]
Please fix the CI format and copyright failure.

**Comment by Wenxing-hou** [comment]
Please refer to other PR to add the issue number in comment.

File: include/library/spdm_crypt_lib.h [hunk 1]
```diff
@@ -893,6 +896,24 @@ bool libspdm_x509_certificate_check(const uint8_t *cert, size_t cert_size,
                                     uint32_t base_asym_algo, uint32_t base_hash_algo,
                                     bool is_requester_cert, bool is_device_cert_model);
 
+
```
Comment (Wenxing-hou) [inline include/library/spdm_crypt_lib.h:899]: There are many extra empty lines in this PR. Please fix them.

**Comment by Wenxing-hou** [comment]
Please change the `1` to `SPDM_DEVICE_CERT_MODEL`.
https://github.com/DMTF/libspdm/blob/2fb7e65a8361818fbb569b7b96a16febb901f68f/os_stub/spdm_device_secret_lib_sample/lib.c#L991-L995

File: include/library/spdm_crypt_lib.h [hunk 2]
```diff
@@ -875,6 +875,9 @@ bool libspdm_get_random_number(size_t size, uint8_t *rand);
 
 #if LIBSPDM_CERT_PARSE_SUPPORT
 
+#define SPDM_ALIAS_CERT_MODEL   0
+#define SPDM_DEVICE_CERT_MODEL  1
```
Comment (jyao1) [inline include/library/spdm_crypt_lib.h:879]: why put here?

File: library/spdm_crypt_lib/libspdm_crypt_cert.c [hunk 3]
```diff
@@ -1177,10 +1177,50 @@ bool libspdm_x509_certificate_check(const uint8_t *cert, size_t cert_size,
                                     bool is_device_cert_model)
 {
     bool status;
+    uint8_t cert_model = SPDM_DEVICE_CERT_MODEL;
+
+    if (!is_device_cert_model) {
+        cert_model = SPDM_ALIAS_CERT_MODEL;
+    }
```
Comment (jyao1) [inline library/spdm_crypt_lib/libspdm_crypt_cert.c:1184]: please use `if () {} else {}` style assignment.

File: library/spdm_crypt_lib/libspdm_crypt_cert.c [hunk 4]
```diff
@@ -1209,17 +1249,55 @@ bool libspdm_x509_set_cert_certificate_check(const uint8_t *cert, size_t cert_si
                                              bool is_requester_cert, bool is_device_cert_model)
 {
     bool status;
+    uint8_t cert_model = SPDM_DEVICE_CERT_MODEL;
+
+    if (!is_device_cert_model) {
+        cert_model = SPDM_ALIAS_CERT_MODEL;
+    }
```
Comment (jyao1) [inline library/spdm_crypt_lib/libspdm_crypt_cert.c:1256]: please use `if () {} else {}` style assignment.

File: include/library/spdm_crypt_lib.h (hunk 2 above)
Comment (sunceping) [reply include/library/spdm_crypt_lib.h]: Now remove it, use the definitions(SPDM_CERTIFICATE_INFO_CERT_MODEL_DEVICE_CERT ..  ) that is already defined in the [spdm.h](https://github.com/DMTF/libspdm/blob/22a9055bf545ed225ad7648ded1decddc47b29a9/include/industry_standard/spdm.h#L493)

File: library/spdm_crypt_lib/libspdm_crypt_cert.c (hunk 3 above)
Comment (sunceping) [reply library/spdm_crypt_lib/libspdm_crypt_cert.c:1185]: Thanks, now fixed.

File: library/spdm_crypt_lib/libspdm_crypt_cert.c (hunk 4 above)
Comment (sunceping) [reply library/spdm_crypt_lib/libspdm_crypt_cert.c:1259]: Thanks, now fixed.

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: include/library/spdm_crypt_lib.h [hunk 5]
```diff
@@ -893,6 +893,23 @@ bool libspdm_x509_certificate_check(const uint8_t *cert, size_t cert_size,
...
 
+/**
+ * Certificate Check for SPDM leaf cert. It is uesd for SPDM 1.3.
+ *
+ * @param[in]  cert                  Pointer to the DER-encoded certificate data.
+ * @param[in]  cert_size             The size of certificate data in bytes.
+ * @param[in]  base_asym_algo        SPDM base_asym_algo
+ * @param[in]  base_hash_algo        SPDM base_hash_algo
+ * @param[in]  is_requester_cert     Is the function verifying requester or responder cert.
+ * @param[in]  cert_model            If 1, the local endpoint uses the DeviceCert model.
```
[the remark this answers is not in the document]
Comment (sunceping) [reply include/library/spdm_crypt_lib.h]: Thanks, it's changed now.

File: library/spdm_crypt_lib/libspdm_crypt_cert.c [hunk 6]
```diff
@@ -1441,6 +1522,70 @@ bool libspdm_get_dmtf_subject_alt_name(const uint8_t *cert, const size_t cert_si
         name_buffer_size, oid, oid_size);
 }
 
+/**
+ * This function verifies the integrity of certificate chain data without spdm_cert_chain_t header.
+ * It is uesd for SPDM 1.3.
```
[the remark this answers is not in the document]
Comment (sunceping) [reply library/spdm_crypt_lib/libspdm_crypt_cert.c]: Thanks, now fixed.

File: library/spdm_crypt_lib/libspdm_crypt_cert.c [hunk 7]
```diff
@@ -1177,10 +1176,52 @@ bool libspdm_x509_certificate_check(const uint8_t *cert, size_t cert_size,
...
+    }
+
+    status = libspdm_x509_common_certificate_check(cert, cert_size, base_asym_algo,
+                                                   base_hash_algo, is_requester_cert,
+                                                   cert_model);
+    if (!status) {
+        return false;
+    }
+
+    /* verify basic constraints: the leaf cert always is ca:fasle in get_cert*/
```
[the remark this answers is not in the document]
Comment (sunceping) [reply library/spdm_crypt_lib/libspdm_crypt_cert.c]: Thanks, now fixed.

File: library/spdm_crypt_lib/libspdm_crypt_cert.c [hunk 8]
```diff
@@ -1018,8 +1017,8 @@ static bool libspdm_verify_leaf_cert_spdm_extension(const uint8_t *cert, size_t
 
     /* Responder does not determine Requester's certificate model */
     if (!is_requester_cert) {
-        if ((find_sucessful) && (!is_device_cert_model)) {
-            /* Hardware_identity_OID is found in alias cert model */
+        if ((find_sucessful) && (cert_model != SPDM_CERTIFICATE_INFO_CERT_MODEL_DEVICE_CERT)) {
```
[the remark this answers is not in the document]
Comment (sunceping) [reply library/spdm_crypt_lib/libspdm_crypt_cert.c]: Thanks, now fixed.

**Review by Wenxing-hou — APPROVED** [review]
LGTM.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**